apps/license-api/src/app/modules/license/license.controller.ts

```diff
@@ -0,0 +1,64 @@
+import { BadRequestException, Body, Controller, Post } from '@nestjs/common'
+import { LicenseService } from './license.service'
+import { Audit } from '@island.is/nest/audit'
+import { Documentation } from '@island.is/nest/swagger'
+import {
+  UpdateLicenseResponse,
+  UpdateLicenseRequest,
+  RevokeLicenseResponse,
+  RevokeLicenseRequest,
+  VerifyLicenseRequest,
+  VerifyLicenseResponse,
+} from './dto'
+import { ApiTags } from '@nestjs/swagger'
+
+@Controller()
+@ApiTags('license-api')
+@Audit()
+export class LicenseController {
```

Comment:
I would refactor the controller to provide more resource based endpoints, more in the direction of restful resource rather than RPC, just make the `verify` a RPC action on the `licenses` resource.

So use the `@Controller` decorator to set `/licenses` as the base route for the controller: `@Controller('/licenses')` or `@Controller({path: '/licenses', version: ['1']})` to enable versioning `/v1/licenses` (that requires `enableVersioning: true` in `boostrap()` in `main.ts`).

Let the `update` method be `@Put(':licenseType/.id')` or `@Patch(':licenseType/.nationalId')` if it is updating partial data of the resource.
 ```
 PUT /v1/licenses/disability/.nationalId
 X-Param-NationalId: 0101307789
```
Then with the dto in the request body, just remove the license type and nationalId from the dto.

Let the `revoke` method be `@Delete(':licenseType/.nationalId')`.
```
DELETE /v1/licenses/disability/.nationalId
X-Param-NationalId: 0101307789
```
With no request body.

Let the `verify` method be `@Post(':licenseType/.nationalId)`.
```
POST /v1/licenses/disability/.nationalId/verify
X-Param-NationalId: 0101307789
```
With the barcode data in the request body.

Nest provides a `@Headers('X-Param-Id')` to get specific header value and `@Param('licenseType')` to get a path param.

We are working on a update on the API Design Guide were we are recommending removing PII info from path in the URL, so monitoring systems aren't logging it. But the idea is that hardcoding the path segment where the PII id would be and use i.e. `.nationalId` and provide the value in a HTTP header that would be named `X-Param-NationalId`

Reply:
Alternatively the paths could be: 

/v1/license-types/:licenseType/licenses/.nationalId

Or maybe even better:

/v1/users/.nationalId/licenses/:licenseType

Reply:
@thorkellmani I updated the comment above, just in case you have already reviewed it.

Reply:
@thorkellmani edited the comment with @eirikurn suggestions.

Reply:
@saevarma since `verify` is only based on the barcode with an ID which is mapped to the license type, would it fit within the guidelines to do a separate route:

```
POST /v1/verify
```

With the barcode data in the request body.

Reply:
Otherwise the changes to the controller looks good. I would think about maybe creating two controllers:
```
@Controller({ version: ['1'], path: 'users/.nationalId/licenses/' })
@ApiTags('users-licenses')
export class UsersLicensesController {
  @Put(':licenseId')
  async update()

  @Delete(':licenseId')
  async revoke()
}

@Controller({ version: ['1'], path: 'licenses/' })
@ApiTags('licenses')
export class LicensesController {
  @Post('/verify')
  async verify()
}
```

Also I would place the `license-api` app inside the `apps/services` directory in the monorepo.